0028-web-monetization/0028-web-monetization.md

```diff
@@ -1,79 +1,296 @@
+| `data` | `Event.data: Uint8Array` | When data is received over the stream. |
+| `money` | `Event.amount: String` | When money is received over the stream. |
+| `outgoing_money` | `Event.amount: String` | When money is sent over the stream. |
+| `close` | | When the stream is closed. |
+| `error` | `Event.error: Error` | When the stream encounters an error. |
+
+#### Fields (Read-only)
+
+| Name | Type | Value |
+|:---|:---|:---|
+| `id` | `String` | The id of this stream, unique to this connection. |
+| `receiveMax` | `String` | The maximum amount you permit this stream to receive. Change with `setReceiveMax()` |
+| `sendMax` | `String` | The amount you will try to send on this stream. Change with `setSendMax()` |
+| `totalReceived` | `String` | The total amount received so far on this stream. |
+| `totalSent` | `String` | The total amount sent so far on this stream. |
+
+#### Check if Stream is Open
+
+```ts
+stream.isOpen(): Boolean
+```
+
+##### Returns
+
+- `Boolean`, whether the stream is open.
+
+#### Allow Money to be Received
+
+```ts
+stream.setReceiveMax(amount: String): void
+```
+
+Sets the maximum that can be received. Must be called with
+increasing values.
+
+##### Parameters
+
+- `amount: String` - The amount to receive (in our units).
+
+#### Wait for Money to be Received
+
+```ts
+stream.receiveTotal(amount: String, timeout?: Number): Promise<void>
+```
+
+Sets the maximum that can be received and waits until that amount has been
+reached. Must be called with increasing `amount`s.
+
+##### Parameters
+
+- `amount: String` - The amount to receive (in our units).
+- `timeout: Number` - How long (in ms) to wait for this function to complete.
+
+#### Try to Send Money
+
+```ts
+stream.setSendMax(amount: String): Promise<void>
+```
+
+Sets the maximum that can be sent. The stream will send as much as the receiver
+is willing to receive, up to this maximum.The stream will send as much as the
+receiver is willing to receive, up to this maximum. Must be called with
+increasing values.
+
+##### Parameters
+
+- `amount: String` - The amount to send (in our units).
+
+#### Wait for Money to be Sent
+
+```ts
+stream.sendTotal(amount: String, timeout?: Number): Promise<void>
+```
+
+Sets the maximum that can be sent and waits until that amount has been
+sent. Must be called with increasing `amount`s.
+
+##### Parameters
+
+- `amount: String` - The amount to send (in our units).
+- `timeout: Number` - How long (in ms) to wait for this function to complete.
+
+#### Send Data on the Stream
+
+```ts
+stream.send(msg: Uint8Array | String): Promise<void>
+```
 
-Rejects with `Error` if the user's browser decides not to send money to this page.
+Sends data over the stream, which will be emitted as `data` events on the other
+side. The data may be broken into chunks or buffered; one `send` call doesn't
+necessarily correspond to a single `data` event.
 
+##### Parameters
+
+- `msg: Uint8Array | String` - The data to send over the stream.
```

Comment:
If the user sends a string, will the other side get a string or just bytes? (I assume it's bytes but it would be worth mentioning)